### PR TITLE
Implements `userinfo` decorator to be used with `discovery` option

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
Implemented a 6th decorator to be used when Authorization Server metadata `discovery` feature is being used.
This helps users of the plugin to automate more.
By using `userinfo` decorator directly, instead of having to construct a call to `userinfo_endpoint` them selves (find, configure urls, provide env variables etc...), 
...they can have less cruft in code 🚂 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by big-kahuna-burger** [comment]
Anyone want to give it a review?
Implemented are 3 variants of fetching userinfo that might be supported by OIDC providers defined in [OIDC Core - Userinfo](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) section.

File: examples/userinfo.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+})
+
+// using async/await (promises API) ->
+// 1. simple one with async
+fastify.get('/interaction/callback/google', async function (request, reply) {
+  try {
+    const tokenResponse = await this.googleOAuth2.getAccessTokenFromAuthorizationCodeFlow(request, reply)
+    const userinfo = await this.googleOAuth2.userinfo(tokenResponse.token /* or tokenResponse.token.access_token */)
+    reply.send(userinfo)
```
Comment (mcollina) [inline examples/userinfo.js:46]:
```suggestion
    return userinfo
```

File: examples/userinfo.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  } catch (error) {
+    // you want to handle error here better
+    console.error(error)
+    reply.send(error.message)
+  }
```
Comment (mcollina) [inline examples/userinfo.js:51]: remove the try/catch, this is not expresa

File: README.md [hunk 3]
```diff
@@ -363,6 +363,38 @@ fastify.googleOAuth2.revokeAllToken(currentAccessToken, undefined, (err) => {
    // Handle the reply here
 });
 ```
+
+- `userinfo(tokenOrTokenSet)`: A function to retrieve userinfo data from Authorization Provider. Both token (as object) or `access_token` string value can be passed.
```
Comment (mcollina) [inline README.md:367]: Can you mention the RFC?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**